### ISSUE TITLE
Setting `specfile` and `drmfile` has no effect when calling `stx_convert_spectrogram`

### ISSUE THREAD

**Issue by settwi** [i4Ds/STIX-GSW#131]
### What is not working?
Setting `specfile` and `drmfile` has no effect when calling `stx_convert_spectrogram`.
The default file names are always used.

### What is the expected behavior?
That setting the `specfile` and `drmfile` arguments change the output spectrum and SRM file names.

## Files
The issue is here: lines 254 and 255 of [stx_convert_science_data2ospex.pro](https://github.com/i4Ds/STIX-GSW/blob/56acbbbe297e14b88c0cd4a17966f8e2d97e6984/stix/idl/processing/spectrogram/stx_convert_science_data2ospex.pro#L254-L255).
The default file name always overwrites the one in `fits_info_params`.

## Potential fix
Change the aforementioned lines to the following
```idl
cur_spec_fn = fits_info_params.specfile
cur_srm_fn = fits_info_params.srmfile
fits_info_params.specfile = (cur_spec_fn eq '') ? specfilename : cur_spec_fn
fits_info_params.srmfile = (cur_srm_fn eq '') ? srmfilename : cur_srm_fn
```

## Example Code
Download L1 or L1A data for observation ID `2204202800` to `2204202800.fits`.
Then run this code:
```idl
    raw_spectrogram_fn = '2204202800.fits'
    stx_get_header_corrections, raw_spectrogram_fn, $
        distance = distance, time_shift = time_shift

    spex_specfile = 'stix_spec_' + raw_spectrogram_fn
    spex_drmfile = 'stix_drm_' + raw_spectrogram_fn

    stx_convert_spectrogram, $
        fits_path_data = raw_spectrogram_fn, $
        distance = distance, $
        time_shift = time_shift, $
        ospex_obj = ospex_obj, $
        specfile = spex_specfile, $
        srmfile = spex_drmfile
```

Files are output with the default names even though specfile and srmfile have been specified.

## Environment
- IDL Version:
```
IDL> help, !version
** Structure !VERSION, 8 tags, length=104, data length=100:
   ARCH            STRING    'x86_64'
   OS              STRING    'linux'
   OS_FAMILY       STRING    'unix'
   OS_NAME         STRING    'linux'
   RELEASE         STRING    '8.7.0'
   BUILD_DATE      STRING    'Jan 22 2018 (347767)'
   MEMORY_BITS     INT             64
   FILE_OFFSET_BITS
                   INT             64
```
- STIX GSW Version: `"The current STIX Ground Software version is - v0.3.1"`
- OS: Linux

**Comment by settwi** [comment]
@grazwegian i sent an email about this earlier but here is a proper github issue

**Comment by settwi** [comment]
i realize now this is a duplicate of #126 ...

**Comment by settwi** [comment]
Fixed by #135
